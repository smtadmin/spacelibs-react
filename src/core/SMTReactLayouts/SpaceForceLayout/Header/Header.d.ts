export default Header;
declare class Header {
    constructor(props: any);
    render(): JSX.Element;
}
declare namespace Header {
    namespace defaultProps {
        const bootstrap: string;
    }
    namespace propTypes {
        const left: any;
        const center: any;
        const right: any;
    }
}
