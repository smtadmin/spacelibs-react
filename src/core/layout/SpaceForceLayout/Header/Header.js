/*
 * File: Header
 * Version: 0.1.0
 * Project: react-test
 * Description: Component that wraps and formats page header content
 * File Created: Friday, 29th January 2021 9:57 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Tuesday, 20th July 2021 2:44 pm
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import * as React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import Col from 'react-bootstrap/Col';
import PropTypes from 'prop-types';

/** @jsx jsx */
import { css, jsx } from "@emotion/react";
import { useTheme } from "@material-ui/core/styles";

/**
 * Header Component
 *
 * @param {*} props Component props
 * @returns {*} React component 
 */
function Header(props) {
	const theme = useTheme();

	return (
        <Navbar
            id='header-container'
            className={props.bootstrap}
            css={css`
                background-color: ${theme.palette.primary.mainOriginal};
            `}>
            <Col className='header-left col-2'>{props.left}</Col>
            <Col className='header-center col-8'>{props.center}</Col>
            <Col className='header-right col-2'>{props.right}</Col>
        </Navbar>
    );
}

/** Sets Default Props */
Header.defaultProps = {
    bootstrap: "header"
};
/** Prop Validators */
Header.propTypes = {
    left: PropTypes.any,
    center: PropTypes.any,
    right: PropTypes.any,
    bootstrap: PropTypes.string
};

export default Header;