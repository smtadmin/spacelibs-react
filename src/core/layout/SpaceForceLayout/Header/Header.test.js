/*
 * File: Header.test
 * Version: 0.1.0
 * Project: react-test
 * Description: Testing Suite for the Header component
 * File Created: Saturday, 6th February 2021 10:55 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Friday, 12th February 2021 9:11 am
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import Header from './Header';
import React from 'react';
import { render } from '@testing-library/react';

/**
 * Attempts to render the header component with a object passed to it
 */
it("Renders without crashing", () => {
    const headerContent = {
        left: "left",
        Center: "center",
        right: "right"
    }
    const element = render(<Header header={headerContent} />)
    expect(element).toBeTruthy();
});

/**
 * Renders the header component and compared it to a snapshot to confirm that output is consistent
 */
it("Renders correctly", () => {
    const headerContent = {
        left: "left",
        Center: "center",
        right: "right"
    }
    const element = render(<Header header={headerContent} />)
    expect(element).toMatchSnapshot();
});

/**
 * Passes arrays of element to each place in the header component and compares it to a snapshot to confirm that the output is consistent
 */
it("Renders multiple items correctly", () => {
    const headerContent = {
        left: ["item1", "item2", "item3"],
        Center: ["item1", "item2", "item3"],
        right: ["item1", "item2", "item3"]
    }
    const element = render(<Header header={headerContent} />)
    expect(element).toMatchSnapshot();
});