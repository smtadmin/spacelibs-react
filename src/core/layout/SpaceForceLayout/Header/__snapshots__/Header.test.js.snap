// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="header css-1ri2fes-Header navbar navbar-expand navbar-light"
        id="header-container"
      >
        <div
          class="header-left col-2 col"
        />
        <div
          class="header-center col-8 col"
        />
        <div
          class="header-right col-2 col"
        />
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="header css-1ri2fes-Header navbar navbar-expand navbar-light"
      id="header-container"
    >
      <div
        class="header-left col-2 col"
      />
      <div
        class="header-center col-8 col"
      />
      <div
        class="header-right col-2 col"
      />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders multiple items correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="header css-1ri2fes-Header navbar navbar-expand navbar-light"
        id="header-container"
      >
        <div
          class="header-left col-2 col"
        />
        <div
          class="header-center col-8 col"
        />
        <div
          class="header-right col-2 col"
        />
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="header css-1ri2fes-Header navbar navbar-expand navbar-light"
      id="header-container"
    >
      <div
        class="header-left col-2 col"
      />
      <div
        class="header-center col-8 col"
      />
      <div
        class="header-right col-2 col"
      />
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
